tests/integration/security/authorization_test.go

```diff
@@ -940,118 +1009,121 @@ func TestAuthorization_Conditions(t *testing.T) {
 			nsB := apps.Namespace2
 			nsC := apps.Namespace3
 
-			cSet := apps.C.Match(echo.Namespace(nsC.Name()))
-
-			var IPC string
-			for i := 0; i < len(t.Clusters()); i++ {
-				IPC += "\"" + getWorkload(cSet[i], t).Address() + "\","
-			}
-			lengthC := len(IPC)
-			IPC = IPC[:lengthC-1]
-			portC := 8090
-			for i := 0; i < len(t.Clusters()); i++ {
-				t.NewSubTest(fmt.Sprintf("IpA IpB IpC in %s", t.Clusters()[i].StableName())).Run(func(t framework.TestContext) {
-					podAWithIPA := apps.A.Match(echo.InCluster(t.Clusters()[i])).Match(echo.Namespace(nsA.Name()))[0]
-					podBWithIPB := apps.B.Match(echo.InCluster(t.Clusters()[i])).Match(echo.Namespace(nsB.Name()))[0]
-
-					args := map[string]string{
-						"NamespaceA": nsA.Name(),
-						"NamespaceB": nsB.Name(),
-						"NamespaceC": nsC.Name(),
-						"IpA":        getWorkload(podAWithIPA, t).Address(),
-						"IpB":        getWorkload(podBWithIPB, t).Address(),
-						"IpC":        IPC,
-						"PortC":      fmt.Sprintf("%d", portC),
-					}
-
-					policies := tmpl.EvaluateAllOrFail(t, args, file.AsStringOrFail(t, "testdata/authz/v1beta1-conditions.yaml.tmpl"))
-					t.Config().ApplyYAMLOrFail(t, "", policies...)
+			c := apps.C.Match(echo.Namespace(nsC.Name()))
+			vm := apps.VM.Match(echo.Namespace(nsA.Name()))
+			for _, dst := range []echo.Instances{c, vm} {
+				for i := 0; i < len(t.Clusters()); i++ {
```

Comment:
it's generally unreliable to loop over `ctx.Clusters` and it's better to loop over `apps.A`. I do however see that you want to get both `a` and `b`. 

Would it be possible to do:

```go
for _, src1 := range apps.A {
  src1, src2  := src1, apps.B.Match(echo.InCluster(src1.Config().Cluster())
  t.NewSubtest(...).Run(func(t framework.TestContext){
    ...
   })
}
```

Reply:
This would help to avoid the `len(from) == 0`. Also I'm sort of confused how that is possible? Do we not deploy the same set of apps consistently across clusters?